### ISSUE TITLE
Original German A2000 & SDRAM Refresh

### ISSUE THREAD

**Issue by jasonsbeer** [jasonsbeer/Amiga-N2630#12]
**Description:** SDRAM will not refresh with the original German A2000. 

**Severity:** Low? I am not sure how many of the original German made A2000's are in the wild and how many of those few will want to use the N2630. Cost reduced versions (B2000) are not affected.

**Observations/Useful Info:** U601 and U602 consume the 7MHz clock to drive the SDRAM refresh timer. The original German A2000 does not supply the 7MHz clock at the coprocessor slot, making it unavailable to the N2630 for this purpose.

**Proposal:** Modify U601 and U602 to consume the 50MHz clock to support the original German A2000.

**Comment by jasonsbeer** [comment]
Upon further consideration, this will require a hardware revision to implement. U601 needs the 7MHz clock to adhear to DMA timing. This clock will be missing on original A2000 machines. It should be possible to bodge this on current Rev 3.x boards. U600 will need a pin assignment update to implement.

**Comment by TheBajaGuy** [comment]
Maybe this will help: You need an XOR of C1 and C3 clocks off the motherboard slot to create your board's 7M clock on all systems.  

It's probably the safer 7M 'clock' to work with, as 7M off the machine slot can sometimes be messy from other system loads.

**Comment by jasonsbeer** [comment]
I looked at recreating the 7MHz system clock from the C1 and C3 clocks. The A500/A2000 Technical Reference recommends the equation "C1 XNOR C3" (pp. 89). XOR would result in the inverse of the 7MHz clock.

Unfortunately, this results in a delay of approximately 30ns between the recreated clock and the original A2000 7MHz clock. Using this recreated clock results in video artifacts. I am unable to determine the exact issue as the signal edges are still within the expected 68000 states. Logic analyzer output attached for future reference.

I am skipping this issue for now (but leaving it open), as there are other more important issues to deal with.

![image](https://github.com/jasonsbeer/Amiga-N2630/assets/78610265/ad3fcbf5-bda5-4f82-9d80-80f455593ada)

**Comment by TheBajaGuy** [comment]
Hi Jason,
Thanks for the analyzer output, and the logic correction.  It's been ~30 years since I worked for GVP Tech Support, and a couple more (while there) since I hacked an A3001/25 to work on an A1000 or A500 expansion edge with 2x 86-pin L slot connectors soldered properly together, plus the synth of that missing clock logic (and created the _BOSS line logic to a pin on the 68000, because that's also missing on that connector on a German A2000).  FWIW - The 50MHz version actually worked fine on the A500, but the A1000's loose tolerances were shaky at 33MHz clocking.  My hacking moment resulted in Jeff Boyer (GVP Engineer) modifying the A2000 G-Force 030 into what became the GVP A530.

From your comment about the artifacts, I was looking at the E-clock, which feeds some motherboard resources (CIA registers, for one).  The output shows the E-clock transition is lining up on the fall of your CLK7-2630.  The native CLK7 signal is always low where the E-clock transitions.  As the E-clock is derivative from the 7M (& normally made in the 68000 when active), maybe it's generation from a different source (and the 30ns skew from the c1/c3 synth) has something to do with it?
Thought: As the German A2000 is kind of a corner case, and you also have the no-fallback to 68000 mode with them, maybe you could address a few issues in one.  An optional under-68000 PCB could have the missing gate needed to create _BOSS, hold the 68000 so it doesn't need to be removed/grant fallback, and you would have access to the 7M clock - if you wanted to steal it.  Possibly a headache to rev the board again to allow a 2-wire connection cleanly, though.
Low priority in any case, as you noted.
Robert




   On Monday, October 2, 2023 at 04:38:43 PM EDT, Jason Neus ***@***.***> wrote:  
 
 


I looked at recreating the 7MHz system clock from the C1 and C3 clocks. The A500/A2000 Technical Reference recommends the equation "C1 XNOR C3" (pp. 89). XOR would result in the inverse of the 7MHz clock.

Unfortunately, this results in a delay of approximately 30ns between the recreated clock and the original A2000 7MHz clock. Using this recreated clock results in video artifacts. I am unable to determine the exact issue as the signal edges are still within the expected 68000 states. Logic analyzer output attached for future reference.

I am skipping this issue for now (but leaving it open), as there are other more important issues to deal with.

**Comment by jasonsbeer** [comment]
> The output shows the E-clock transition is lining up on the fall of your CLK7-2630.  The native CLK7 signal is always low where the E-clock transitions.  As the E-clock is derivative from the 7M (& normally made in the 68000 when active), maybe it's generation from a different source (and the 30ns skew from the c1/c3 synth) has something to do with it?>

Could be. The data is latched on the falling edge of 68000 State 6 for reads from the A2000 board. This event is supposed to be synchronous with the falling edge of E. But, as you point out, there is some variance around that when the clocks are derived from different sources. The way to test this theory is to pull the 68000 and put it in "A"2000 mode. In that mode, the 2630 generates its own E signal based on the recreated clock. That should align everything. Something I will try in the near future.

I enjoyed reading about your GVP experiences. As a young man at the time, I felt anybody who was somebody purchased GVP peripherals. Although, I had to settle for lesser brands on the used market. 8^>

**Comment by TheBajaGuy** [comment]
After I started at GVP, I was introduced to the person who ran the Philadelphia Amiga Users Group's 5-line BBS - an A2000 stuffed with our top gear ('040, serial cards, 'spare' hard disks), and he became one of my friends.  He referred to my job as working in the 'GVP Toy Factory' - and looking back, it kind of was.  While I wasn't paid all that much during that time of employ - I was 20 when I started - there was an education and experience to be had that no school could have ever provided.  It was a frontier, of sorts.  The 'wild wild west' of computing.  The job had lots of stress, but it was also lots of fun.

I dropped the last run you posted into Paint, zoomed in, and there are points where the synth CLK7-A2630 is still high as E is falling (4th red line, pasted below as an example), but the CLK7 is clearly low.  I eye-balled the rest of the capture, and it looks like several others are lined up like much that one.  

I was updating the BBOAH's Amiga 2000 page in the past week, and it reminded me of Dave Haynie mentioning some changes he made for the B2000, including how buffering around the CPU slot, particularly clocks, is more beefed up than on the A2000.  I don't recall all the details of the (first) GVP A3001, but there were 4 jumpers to change on it when used in the German A2000, and the 68000 was of course removed.  I never had the schematics, but the J12/13/14 changes I believe are related to the C1/C3 (connect) and 7M (disconnect).  I think J10 does something to compensate for the missing 68000.  The other reason I mention this is the GVP A3001 front card also could use the C= ROMs for Unix or Boot Selection, and we had an IDE interface, too.  Sadly, our board was out before the finalized the AT-IDE standard.  The final timing on ours was off frm that spec.  GVP was quite small at the time - I was the 9th employee hired in late 1989 - so another board rev was not an option for quite awhile.  The drives that did work dried up by late 1990, and so the Combo cards with DPRC DMA SCSI became the next generation.

I wish I had the time to get one of these boards built and give it a run.  I have quite a stack of GVP and other Z2/Z3 cards in my collection.  I participate in the Amiga OS Beta Test group (3.1.4, 3.2, etc) as time allows.  I use my 'collection' to test things like AutoConfig behavior (in the OS) with fast accelerators.  I also have produced the GuruROM v6 (for GVP and A2091/590) from Ralph Babel.  Unfortunately, I'm very behind on producing them for the waiting list on Aminet - life happens, personal priorities, and all that, as I am sure you know.

![image](https://github.com/jasonsbeer/Amiga-N2630/assets/87320034/b61586de-d257-462b-a6c3-6ecfaa19aa39)
